show-line-of-sight-between-geoelements/src/main/java/com/esri/arcgismaps/sample/showlineofsightbetweengeoelements/components/SceneViewModel.kt

```diff
@@ -0,0 +1,259 @@
+import androidx.lifecycle.viewModelScope
+import com.arcgismaps.Color
+import com.arcgismaps.analysis.GeoElementLineOfSight
+import com.arcgismaps.geometry.AngularUnit
+import com.arcgismaps.geometry.GeodeticCurveType
+import com.arcgismaps.geometry.GeometryEngine
+import com.arcgismaps.geometry.LinearUnit
+import com.arcgismaps.geometry.Point
+import com.arcgismaps.geometry.PointBuilder
+import com.arcgismaps.geometry.SpatialReference
+import com.arcgismaps.mapping.ArcGISScene
+import com.arcgismaps.mapping.ArcGISTiledElevationSource
+import com.arcgismaps.mapping.BasemapStyle
+import com.arcgismaps.mapping.Surface
+import com.arcgismaps.mapping.Viewpoint
+import com.arcgismaps.mapping.layers.ArcGISSceneLayer
+import com.arcgismaps.mapping.symbology.ModelSceneSymbol
+import com.arcgismaps.mapping.symbology.SceneSymbolAnchorPosition
+import com.arcgismaps.mapping.symbology.SimpleMarkerSymbol
+import com.arcgismaps.mapping.symbology.SimpleMarkerSymbolStyle
+import com.arcgismaps.mapping.symbology.SimpleRenderer
+import com.arcgismaps.mapping.view.AnalysisOverlay
+import com.arcgismaps.mapping.view.Camera
+import com.arcgismaps.mapping.view.Graphic
+import com.arcgismaps.mapping.view.GraphicsOverlay
+import com.arcgismaps.mapping.view.SurfacePlacement
+import com.esri.arcgismaps.sample.sampleslib.components.MessageDialogViewModel
+import com.esri.arcgismaps.sample.showlineofsightbetweengeoelements.R
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+import java.io.File
+import java.util.Timer
+import java.util.TimerTask
+
+class SceneViewModel(private var application: Application) : AndroidViewModel(application) {
+
+
+    // Create a scene and add a basemap to it
+    val scene = ArcGISScene(BasemapStyle.ArcGISTopographic)
+
+    // Create graphic overlay to hold graphics
+    val graphicsOverlay = GraphicsOverlay()
+
+    // Create an analysis overlay to hold the line of sight
+    val analysisOverlay = AnalysisOverlay()
+
+    // Initialize z to 50 as starting point and observe any changes to its value
+    private val _currentZValue = MutableStateFlow(50.0)
+    val currentZValue: StateFlow<Double> = _currentZValue.asStateFlow()
+
+    // The following variables keep track of the taxi's location
+    private var waypointsIndex = 0
+    private val meters = LinearUnit.meters
+    private val degrees = AngularUnit.degrees
+
+    // Create waypoints around a block for the taxi to drive to
+    private val wayPoints = listOf(
+        Point(-73.984513, 40.748469, SpatialReference.wgs84()),
+        Point(-73.985068, 40.747786, SpatialReference.wgs84()),
+        Point(-73.983452, 40.747091, SpatialReference.wgs84()),
+        Point(-73.982961, 40.747762, SpatialReference.wgs84()),
+    )
+
+    // Create a graphic of a taxi to be the target
+    private val provisionPath: String by lazy {
```

Comment:
The comment here doesn't seem right

```suggestion
    private val provisionPath: String by lazy {
```